src/AutoPilotPlugins/PX4/PX4AutoPilotPlugin.cc

```diff
@@ -94,6 +95,12 @@ const QVariantList& PX4AutoPilotPlugin::vehicleComponents(void)
                 _tuningComponent->setupTriggerSignals();
                 _components.append(QVariant::fromValue((VehicleComponent*)_tuningComponent));
 
+                if(_vehicle->parameterManager()->parameterExists(_vehicle->id(), "SYS_VEHICLE_RESP")) {
+                    _flightBehavior = new PX4FlightBehavior(_vehicle, this);
```

Comment:
You are creating a dangling pointer, be sure to delete it or use a smart pointer (such as QSharedPointer), the first option is the desired way to accomplish it IMO.

Reply:
Yes, memory leak, which affects all other allocations as well. Fixed.